File: array/1260_shift_2d_grid/1260_shift_2d_grid.go
package main

import "fmt"

func shiftGrid(grid [][]int, k int) [][]int {
	res := make([]int, 0, 10)
	lenth := len(grid)
	w := len(grid[0])
	for i := 0; i < lenth; i++ {
		res = append(res, grid[i]...)
	}
	k = k % (lenth * w)
	res = append(res[lenth*w-k:], res[:lenth*w-k]...)
	tmp := make([][]int, 0, 10)
	for i := 0; i < lenth; i++ {
		tmp = append(tmp, res[i*w:(i+1)*w])
	}
	return tmp
}

func main() {
	grid := [][]int{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}}
	res := shiftGrid(grid, 1)
	fmt.Println(res)
}

File: array/1260_shift_2d_grid/1260_shift_2d_grid.py
from typing import List


class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        num_rows, num_cols = len(grid), len(grid[0])

        for _ in range(k):

            previous = grid[-1][-1]
            for row in range(num_rows):
                for col in range(num_cols):
                    temp = grid[row][col]
                    grid[row][col] = previous
                    previous = temp
        return grid


grid = [[1,2,3],[4,5,6],[7,8,0]]
res = Solution().shiftGrid(grid, 1)
print(res)